Add pid_suffix to .imap_notifier config to allow multiple notifiers.

djak250 opened the pull request:
Some users may have multiple emails, and this change allows them to
have a separate imap notifier for each of them, based on a suffix added
to the .pid file.
ex. pid_suffix = "gmail", creates /tmp/image_notifier_gmail.pid.
On `imap_notifier -k`, if multiple imap_notifiers exist, they are
listed so the user can select which one to kill.

djak250 committed:
Add pid_suffix to .imap_notifier config to allow multiple notifiers.

Some users may have multiple emails, and this change allows them to
have a separate imap notifier for each of them, based on a suffix added
to the .pid file.
ex. pid_suffix = "gmail", creates /tmp/image_notifier_gmail.pid.
On `imap_notifier -k`, if multiple imap_notifiers exist, they are
listed so the user can select which one to kill.

djak250 commented:
Let me know if you'd like to me update the README.

djak250 committed:
Fixed nuby ruby mistakes
-Use snake_casing on variables instead of camelCasing.
-Better utilize string interpolation to clean up pid_file creation

djak250 commented:
Yeah you can tell I haven't used ruby in quite some time. Lol.
Small addition: 
This is camelCasing.
This is snake_casing. Haha.

djak250 commented:
One other thing that might make this not ready for merging:
The check for the PID file will fail to find a custom named one. So there has to be some way to know there's going to be a custom pid suffix in the bin/imap_notifier file. Right now it checks for the default `/tmp/imap_notifier.pid`, which doesn't get created if they use a custom suffix. I'll look into this a little later tonight, but would you be against reading in the yaml in that file? I mean, we could just ditch it from the configuration and have it be a command line option instead. Up to you, though

dacamp commented:
> This is camelCasing.
> This is snake_casing. Haha.
:hushed: this fully classifies as a brain spasm... Thanks for correcting me. :)

I haven't been using Ruby lately either, so I can easily be persuaded to go either way, but I would like to maintain some consistent readability.

> The check for the PID file will fail to find a custom named one.
This one is totally on me, but there *really* needs to be tests on this repo.  I doubt I'll get around to it before you've got this ready to merge, but if you have a way you're currently testing this, let me know.

Thanks again for working on this!

djak250 commented:
I've just been testing by trial and error. Haha. I did manage to get a local branch running with the checks for custom suffix pid files, but it mainly just involved me reading in the configuration file again in the bin startup. That was done more to just get it working personally. But if you wouldn't be opposed to reading in the options at that point, rather than in the application initialization, it would take out some code duplication. Up to you how to proceed though. I'll push up my changes to my fork so you can check it out and cringe. ;)

djak250 committed:
Duplicated configuration code to read in on appliction start as well to check against the custom suffix used for a pid file.

djak250 committed:
Moved the file checking to the initialize function.
This change allows only a single read of the config file.
There's been another attribute that can be added to the configuration file called ignore_exit_code. This change allows the imap_notifier command to be chained with another shell command even thought it's already running. This is so that someone may start up mutt with an alias that calls imap_notifier beforehand.

Commit:
Merge pull request #1 from djak250/suffix_pid_check

Suffix pid check

djak250 commented:
So I'm all for doing stuff in the map_notifier.d folder. It would make things a little cleaner looking when we use the -k flag, because you can just list the folder's contents, rather than doing that odd `dir[]` matching command. If you're talking full rewrite though, I'd like to give this a shot at rewriting it using GoLang. For one, I don't think the user would need Ruby (or GoLang in this case) to run the executable in that case. Plus, GoLang kinda forces you to write code in a dry way, or at least doesn't let you compile with any unused fluff.

dacamp commented:
:+1:  Let's redo it in Go, I'm so down!!

I've created an organization and added you.  We'll probably need to hash out details, such as executable delivery.

djak250 commented:
I've actually already got a basic implementation done specifically for my use cases. I'll see about transferring it to this org.
On 08/05, David Campbell wrote:
>:+1:  Let's redo it in Go, I'm so down!!
>
>I've created an organization and added you.  We'll probably need to hash out details, such as executable delivery.
>
>---
>Reply to this email directly or view it on GitHub:
>https://github.com/dacamp/imap_notifier/pull/9#issuecomment-128142754

-- 
David Johnson

matthewp commented:
What is the status of this PR?  I'd really like to use it.  If you're going to rewrite, why not release this first and then work on that?

matthewp commented:
cc @djak250 @dacamp it would be great if this PR could be merged in before a rewrite happens.

dacamp commented:
@matthewp this commit won't work "as-is".  I'll merge it though and update `bin/imap_notifier` opts.  You could also just checkout this commit and run it yourself, but that's also fairly lame...  ¯\_(ツ)_/¯

Sorry for the delay.
